lib/hanami/action/application_action.rb

```diff
@@ -105,6 +97,12 @@ def view_options(req, res)
         def view_context_options(req, res)
           {request: req, response: res}
         end
+
+        # Automatically render the view, if the body hasn't been populated yet
+        def finish(req, res, halted)
+          res.render(view, **req.params, **res.exposures) if view && res.body.empty?
```

Comment:
@cllns As next step, I would extract a _public API_ method:

```ruby
def render?(res)
  view && res.body.empty?
end
```

Pros:

  * Clarity of code
  * Let final users to override this method